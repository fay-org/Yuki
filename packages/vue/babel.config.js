module.exports = {
  presets: ['@fect-ui/cli/preset'],
}
