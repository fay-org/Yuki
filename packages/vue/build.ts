import { BuildTaskConfig, TASK_NAME, runTask, commonOutput } from 'internal'
import { createBundle } from 'no-bump'

const { build } = createBundle({})

const commonInput = 'src/index.ts'

/**
 * I decide split the build task into four.
 * First is only bundle script module.
 * Second part only parser style moudle.
 * Thrid part will generator all analyzed and generator style deps graph.
 * Fourth part will generator the entry file as umd.
 *
 * The first task will have two sub task. with esm task coomonjs task
 */
