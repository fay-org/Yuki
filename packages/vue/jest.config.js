module.exports = {
  testPathIgnorePatterns: ['/node_modules/', '/dist/', '/lib/', '/es/'],
  collectCoverageFrom: [
    'src/**/*.{js,jsx,ts,tsx,vue}',
    '!src/utils/**',
    '!src/index.ts',
    '!src/**/type.ts',
    '!src/**/style.ts',
    '!src/**/interface.ts',
    '!src/back-top/**'
  ],
  collectCoverage: true,
  preset: '@fect-ui/cli/lib/config',
  testTimeout: 20000
}
