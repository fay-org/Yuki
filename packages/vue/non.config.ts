import path from 'path'
import { defineNonConfig } from '@fect-ui/cli'

export default defineNonConfig({
  lib: {
    input: path.join(__dirname, 'components'),
    format: ['cjs', 'es', 'umd'],
    name: 'fect'
  }
})
