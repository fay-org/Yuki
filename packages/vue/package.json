{
  "name": "@fect-ui/vue",
  "version": "1.4.0-rc.1",
  "description": "Minimalist UI components built on Vue",
  "main": "lib/index.js",
  "module": "es/index.js",
  "types": "types/index.d.ts",
  "jsdelivr": "lib/fect.min.umd.js",
  "scripts": {
    "collect": "non collect",
    "build": "npm run collect && non compile && non volar && node ./typing-template.js",
    "test": "non test"
  },
  "keywords": [
    "vue component",
    "fect",
    "vue ui"
  ],
  "homepage": "https://github.com/fect-org/fect",
  "repository": {
    "type": "git",
    "url": "https://github.com/fect-org/fect"
  },
  "files": [
    "lib/",
    "es/",
    "types/",
    "vetur/",
    "README.md",
    "components.d.ts"
  ],
  "author": "XeryYue",
  "license": "MIT",
  "devDependencies": {
    "@fect-ui/cli": "^1.0.0"
  },
  "sideEffects": [
    "es/**/style/*",
    "lib/**/style/*",
    "*.css"
  ],
  "dependencies": {
    "@fect-ui/themes": "^2.2.0",
    "@fect-ui/vue-hooks": "^1.2.0",
    "proy": "^0.0.3",
    "@popperjs/core": "^2.11.2"
  },
  "peerDependencies": {
    "vue": "^3.0.11"
  }
}
