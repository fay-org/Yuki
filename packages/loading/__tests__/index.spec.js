import { mount } from '@vue/test-utils'
import { ref } from 'vue'
import Loading from '../index'
describe('Loading', () => {
  it('should be render as element', () => {
    const loading = mount(Loading)
    expect(loading.html()).toMatchSnapshot()
    expect(() => loading.unmount()).not.toThrow()
  })

  it('should be support types', () => {
    const wrapper = mount({
      setup() {
        return { typeList: ['default', 'success', 'warning', 'error'] }
      },
      render() {
        return (
          <>
            {this.typeList.map((item, i) => (
              <Loading type={item} key={i} />
            ))}
          </>
        )
      },
    })
    expect(wrapper.html()).toMatchSnapshot()
    expect(wrapper.findAll('.fect-loading').length).toEqual(4)
  })

  it('should be support different sizes', () => {
    const wrapper = mount({
      setup() {
        return { sizeList: ['mini', 'small', 'medium', 'large'] }
      },
      render() {
        return (
          <>
            {this.sizeList.map((item, i) => (
              <Loading size={item} key={i} />
            ))}
          </>
        )
      },
    })
    expect(wrapper.html()).toMatchSnapshot()
    expect(wrapper.findAll('.fect-loading').length).toEqual(4)
  })

  it('should be support use custom color', () => {
    const wrapper = mount({
      setup() {
        const cusColor = ref('pink')
        return { cusColor }
      },
      render() {
        return (
          <>
            <Loading color={this.cusColor} />
          </>
        )
      },
    })
    const el = wrapper.find('i')
    expect(wrapper.html()).toMatchSnapshot()
    expect(wrapper.vm.cusColor).toEqual('pink')
    expect(el.attributes('style')).toBe('background: pink;')
  })
  it('should be support loadType', () => {
    const cube = mount(Loading, {
      props: {
        loadType: 'cube',
      },
    })
    const wave = mount(Loading, {
      props: {
        loadType: 'wave',
      },
    })
    expect(cube.html()).toMatchSnapshot()
    expect(cube.findAll('.loading__cube')).toBeTruthy()
    expect(wave.html()).toMatchSnapshot()
    expect(wave.findAll('.loading__wave')).toBeTruthy()
  })
})
