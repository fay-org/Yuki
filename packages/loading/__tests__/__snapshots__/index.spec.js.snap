// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading should be render as element 1`] = `<div class="fect-loading"><span class="loading"><i class="default medium"></i><i class="default medium"></i><i class="default medium"></i></span></div>`;

exports[`Loading should be support different sizes 1`] = `
<div class="fect-loading"><span class="loading"><i class="default mini"></i><i class="default mini"></i><i class="default mini"></i></span></div>
<div class="fect-loading"><span class="loading"><i class="default small"></i><i class="default small"></i><i class="default small"></i></span></div>
<div class="fect-loading"><span class="loading"><i class="default medium"></i><i class="default medium"></i><i class="default medium"></i></span></div>
<div class="fect-loading"><span class="loading"><i class="default large"></i><i class="default large"></i><i class="default large"></i></span></div>
`;

exports[`Loading should be support loadType 1`] = `<div class="fect-loading"><span class="loading"><i class="loading__cube default medium"></i><i class="loading__cube default medium"></i><i class="loading__cube default medium"></i><i class="loading__cube default medium"></i></span></div>`;

exports[`Loading should be support loadType 2`] = `<div class="fect-loading"><span class="loading"><i class="loading__wave default medium"></i><i class="loading__wave default medium"></i><i class="loading__wave default medium"></i><i class="loading__wave default medium"></i><i class="loading__wave default medium"></i></span></div>`;

exports[`Loading should be support types 1`] = `
<div class="fect-loading"><span class="loading"><i class="default medium"></i><i class="default medium"></i><i class="default medium"></i></span></div>
<div class="fect-loading"><span class="loading"><i class="success medium"></i><i class="success medium"></i><i class="success medium"></i></span></div>
<div class="fect-loading"><span class="loading"><i class="warning medium"></i><i class="warning medium"></i><i class="warning medium"></i></span></div>
<div class="fect-loading"><span class="loading"><i class="error medium"></i><i class="error medium"></i><i class="error medium"></i></span></div>
`;

exports[`Loading should be support use custom color 1`] = `<div class="fect-loading"><span class="loading"><i class="default medium" style="background: pink;"></i><i class="default medium" style="background: pink;"></i><i class="default medium" style="background: pink;"></i></span></div>`;
